app/jobs/subscription_confirm_job.rb

```diff
@@ -50,6 +50,8 @@ def confirm_order!(order)
     else
       send_failed_payment_email(order)
     end
+  rescue StandardError => e
+    Bugsnag.notify(e, order: order)
```

Comment:
Re #4981 we already have a bugsnag errors when the processing fails :point_right: https://app.bugsnag.com/yaycode/openfoodnetwork-uk/errors/5eb1a9a5e67a47001777d396?filters[event.since][0][type]=eq&filters[event.since][0][value]=30d&filters[error.status][0][type]=eq&filters[error.status][0][value]=open&filters[search][0][type]=eq&filters[search][0][value]=delayed. What we need to do is rescue the `StateMachine::InvalidTransition` error and then execute `send_failed_payment_email(order)`. 

However, this won't be enough because `SubscriptionConfirmJob#process_payment!` and `Spree::Order#process_payment!` can both return boolean or raise. Therefore, we need to keep that `else` branch above. Yes, those methods need to reduce their complexity. It's a pain in the ass to work with such broad interface/return types. And we're paying its costs already.

Reply:
the objective of this statement is not to generate the bugsnag, the bugsnag is already there but it was blowing up the process, now the bugsnag will be generated and the process will continue to the next order. right?

Reply:
I think we need to rescue StandardError/any exception, otherwise it will blow up. I think I am answering the following comment already: I dont think we need to have a worker for each order, that sounds like over-engineering for this problem. This simple rescue StandardError for each order already does the job of continuing processing the following orders. Let's discuss this tomorrow.

I agree we need to send the failed email anyway :+1: I added a new commit to do that, please have a look.

Reply:
For context: we did a pair programming session and we agreed persisting summaries and making each order being processed by a separate job is a good idea. We will get there. For now we can go with this quicker approach of adding the rescue statements.